### PR TITLE
[cpp-package] add lr scheduler

### PR DESCRIPTION
**Pull request by CNevd**
1. add lr scheduler to optimizer
2. move learning rate and weight decay to optimizer
3. fix examples
@lx75249

### CODE REVIEW

**Comment by lx75249** [comment]
why removing cpu version.. it's included as an example in the tutorial..

**Comment by CNevd** [comment]
sorry will add it back

2017-06-30 11:40 GMT+08:00 Xin Li <notifications@github.com>:
